File: sprint5/By languages/python/lesson8code2.py
def small_left_rotation(a):
  # Задаём обозначения.
  b = a.right
  C = b.left

  # Переусыновляем вершины.
  a.right = C
  b.left = a

  # Корректируем высоты в зависимости от того, равны ли высоты C и R.
  if C.height == R.height:
    a.height -= 1
    b.height += 1
  else:
    a.height -= 2

  # Возвращаем новый корень.
  return b

File: sprint5/By languages/swift/lesson8code2.swift
func smallLeftRotation() -> Node<T> {
    // Задаём обозначения.
    guard let b = right else {
        fatalError("Unable to perform small left rotation: right child is nil.")
    }
    let C = b.left
    
    // Переусыновляем вершины.
    right = C
    b.left = self
    
    // Корректируем высоты в зависимости от того, равны ли высоты C и R.
    if let CHeight = C?.height, CHeight == height {
        height -= 1
        b.height += 1
    } else {
        height -= 2
    }
    
    // Возвращаем новый корень.
    return b
}